Examples/checkers.lisp

```diff
@@ -0,0 +1,446 @@
+   (queenp :initarg :queen  :accessor queenp)
+   (field  :initarg :field  :accessor field))
+  (:default-initargs :queen nil))
+
+(defclass field ()
+  ((col :initarg :col :reader col)
+   (row :initarg :row :reader row)
+   (checker :initform nil :accessor checker)))
+
+(defmethod (setf checker) :before ((val checker) (field field))
+  (setf (checker (field val)) nil
+        (field val) field))
+
+(defmethod (setf checker) :before ((val null) (field field))
+  (alexandria:when-let ((checker (checker field)))
+    (setf (field checker) nil)))
+
+(defclass game-state ()
+  ((board  :initarg :board  :reader board)
+   (player :initarg :player :accessor player)
+   (locked :initarg :locked :accessor locked
+           :documentation "Checker locked to take others.")
+   (over-field   :initarg :over-field   :accessor over-field)
+   (active-field :initarg :active-field :accessor active-field)
+   (game-over :initarg :game-over :accessor game-over)
+   (%takesp :reader takesp))
+  (:default-initargs :board (make-default-board)
+                     :player :player-1
+                     :locked nil
+                     :over-field nil
+                     :active-field nil
+                     :game-over nil))
+
+(defun switch-player (game-state)
+  (setf (locked game-state) nil
+        (active-field game-state) nil)
+  (ecase (player game-state)
+    (:player-1 (setf (player game-state) :player-2))
+    (:player-2 (setf (player game-state) :player-1))))
+
+(defun new-game (game-state)
+  (with-slots (board player locked over-field active-field game-over)
+      game-state
+    (setf board (make-default-board)
+          player :player-1
+          locked nil
+          over-field nil
+          active-field nil
+          game-over nil)
+    (slot-makunbound game-state '%takesp)))
+
+(defun get-field (game-state row col)
+  (unless (integerp row)
+    (setf row (position row "ABCDEFGH" :test #'string=)))
+  (let ((array (board game-state)))
+    (when (array-in-bounds-p array row col)
+      (aref array row col))))
+
+(defun preys (game-state checker)
+  "Returns a list of prey checkers (or NIL) in each direction."
+  (let ((row (row (field checker)))
+        (col (col (field checker))))
+    (flet ((find-prey (rd cd)
+             (do* ((row  #1=(+ row rd)                     #1#)
+                   (col  #2=(+ col cd)                     #2#)
+                   (this #3=(get-field game-state row col) #3#)
+                   (next #4=(get-field game-state #1# #2#) #4#)
+                   (prey #5=(and this (checker this))      #5#)
+                   (blck #6=(or (not next) (checker next)) #6#))
+                  ((null this))
+               (when prey
+                 (if (and (not blck)
+                          (not (eql (player checker)
+                                    (player prey))))
+                     (return-from find-prey prey)
+                     (return-from find-prey nil)))
```

Comment:
The `(return-from find-prey` part can be pulled out of the `if` to indicate that it happens in any case.

Reply:
that certainly looks uglier. nevertheless fixed.